core/dbt/parser/schema_test_builders.py

```diff
@@ -224,13 +224,20 @@ def __init__(
         groups = match.groupdict()
         self.name: str = groups['test_name']
         self.namespace: str = groups['test_namespace']
-        self.modifiers: Dict[str, Any] = {}
-        for key in self.MODIFIER_ARGS:
+        self.config: Dict[str, Any] = {}
+
+        for key in self.CONFIG_ARGS:
             value = self.args.pop(key, None)
+            # 'modifier' config could be either top level arg or in config
+            if not value and 'config' in self.args:
+                value = self.args['config'].pop(key, None)
             if isinstance(value, str):
                 value = get_rendered(value, render_ctx, native=True)
             if value is not None:
-                self.modifiers[key] = value
+                self.config[key] = value
```

Comment:
Could we raise a validation error if the same key exists in `config:` and in top-level arguments?
```yml
- accepted_values:
    values: [1,2]
    where: "2=2"
    config:
      where: "1=1"
```